### DISCUSSION TITLE
Build ekuiper with Plugins

### DISCUSSION THREAD

**Discussion by robodevchampion** [lf-edge/ekuiper#3064] [Q&A] [answered]
Hi, 

I want to build ekuiper docker image with some plugins (Like Kinesis & SQL) so that I can deploy it to multiple edges, tried below Dockerfile 

```
FROM lfedge/ekuiper:latest
WORKDIR /kuiper

COPY artifacts/plugin.so  /kuiper/plugins/sinks/
COPY  artifacts/plugin.json  /kuiper/etc/

USER root
RUN apk add gcompat
RUN ln -s /lib/libgcompat.so.0 /usr/lib/libresolve.so.2

EXPOSE 9099
CMD ["bin/kuiperd"]
```

while after this plugin is visible inside plugin section while it is not visible under actions, Can you please guide on this or may be if possible you can share commands to create plugin via CLI.

Thanks

**Comment by ngjaying** [comment, marked as answer]
Should be `COPY  artifacts/plugin.json  /kuiper/etc/sinks`. Your plugin name is "plugin", that's not very infromative.
